From 50bbdbd499aa995f9d250dd0cef1931e1380a3eb Mon Sep 17 00:00:00 2001
From: Solly Ross <sross@redhat.com>
Date: Tue, 16 Dec 2014 15:42:00 -0500
Subject: [PATCH] Support Neutron

The network driver and network label regex need to be set properly
for neutron to work.

Change-Id: I9e24ea252f434209ebc2fce89727db9bae2d1c1e
---
 trove/manifests/taskmanager.pp | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/trove/manifests/taskmanager.pp b/trove/manifests/taskmanager.pp
index 405df9a..ebbc15a 100644
--- a/trove/manifests/taskmanager.pp
+++ b/trove/manifests/taskmanager.pp
@@ -170,6 +170,32 @@ class trove::taskmanager(
     }
   }
 
+  if $::trove::use_neutron {
+    trove_config {
+      'DEFAULT/network_label_regex':         value => '.*';
+      'DEFAULT/network_driver':              value => 'trove.network.neutron.NeutronDriver';
+    }
+
+    trove_taskmanager_config {
+      'DEFAULT/network_label_regex':         value => '.*';
+      'DEFAULT/network_driver':              value => 'trove.network.neutron.NeutronDriver';
+    }
+  } else {
+    trove_config {
+      'DEFAULT/network_label_regex':         value => '^private$';
+      'DEFAULT/network_driver':              value => 'trove.network.nova.NovaNetwork';
+    }
+
+    trove_taskmanager_config {
+      'DEFAULT/network_label_regex':         value => '^private$';
+      'DEFAULT/network_driver':              value => 'trove.network.nova.NovaNetwork';
+    }
+  }
+
+  trove_config {
+    'DEFAULT/taskmanager_queue':             value => 'taskmanager';
+  }
+
   # Logging
   if $log_file {
     trove_taskmanager_config {
